lib/makeCmdTasks.js

```diff
@@ -31,20 +32,47 @@ export const makeCmdTasks = async ({ commands, cwd, files, gitDir, shell, verbos
     const resolvedArray = Array.isArray(resolved) ? resolved : [resolved] // Wrap non-array command as array
 
     for (const command of resolvedArray) {
-      // If the function linter didn't return string | string[]  it won't work
+      // If the function linter didn't return string | string[] | object it won't work
       // Do the validation here instead of `validateConfig` to skip evaluating the function multiple times
-      if (isFn && typeof command !== 'string') {
+      if ((isFn && typeof command !== 'string' && typeof command !== 'object') || !command) {
         throw new Error(
           configurationError(
             '[Function]',
-            'Function task should return a string or an array of strings',
+            'Function task should return a string or an array of strings or an object',
             resolved
           )
         )
+      } else if (isFn && typeof command === 'object') {
+        if (
+          typeof command.title === 'string' &&
+          typeof command.command === 'string' &&
+          typeof command.task === 'function'
+        ) {
+          const task = async (ctx = getInitialState()) => {
+            try {
+              await command.task()
+            } catch (e) {
+              throw makeErr(command.title, e, ctx)
+            }
+          }
+          cmdTasks.push({
+            title: command.title,
+            command: command.command,
```

Comment:
Do we need both the `command` and `title`?

Reply:
I think we only need title, so removed the command field.